### ISSUE TITLE
Terraform provider does not support BQ's native schema update behavior. Specifically remove column operations

### ISSUE THREAD

**Issue by red8888** [terraform-google-modules/terraform-google-bigquery#145]
### TL;DR

The terraform provider should behave like native BQ does wrt schema updates: https://cloud.google.com/bigquery/docs/managing-table-schemas

It currently does not. Remove column operations are treated destructively- terraform wants to completely destroy and recreate the table.

### Expected behavior

I modify a schema to remove a column.

Terraform updates in place, just like it currently does for add column operations.

### Observed behavior

I modify a schema to remove a column.

Terraform wants to destroy the entire table and recreate it.

### Terraform Configuration

```hcl
# original config
resource "google_bigquery_table" "default" {
  dataset_id = google_bigquery_dataset.default.dataset_id
  table_id   = "bar"
  schema = <<EOF
[
  {
    "name": "removeme",
    "type": "BOOLEAN",
    "mode": "NULLABLE",
    "description": "I should be able to remove this"
  },
  {
    "name": "permalink",
    "type": "STRING",
    "mode": "NULLABLE",
    "description": "The Permalink"
  },
  {
    "name": "state",
    "type": "STRING",
    "mode": "NULLABLE",
    "description": "State where the head office is located"
  }
]
EOF
}

# remove column removeme and balk at terraform's unmitigated audacity
resource "google_bigquery_table" "default" {
  dataset_id = google_bigquery_dataset.default.dataset_id
  table_id   = "bar"
  schema = <<EOF
[
  {
    "name": "permalink",
    "type": "STRING",
    "mode": "NULLABLE",
    "description": "The Permalink"
  },
  {
    "name": "state",
    "type": "STRING",
    "mode": "NULLABLE",
    "description": "State where the head office is located"
  }
]
EOF
}
```

### Terraform Version

```sh
Terraform v0.14.4
+ provider registry.terraform.io/hashicorp/google v3.67.0
+ provider registry.terraform.io/hashicorp/google-beta v3.67.0
```

### Additional information

A user already opened this issue, but for some reason it was closed: https://github.com/terraform-google-modules/terraform-google-bigquery/issues/119

**Comment by morgante** [comment]
Hi, issues with the Terraform provider should be reported on that repo: https://github.com/hashicorp/terraform-provider-google

This repo is for the BigQuery module, which you are not using (as evident from your code sample).

**Comment by danielcampagnolitg** [comment]
@red8888  This is the issue in the underlying terraform provider you'll want to follow
https://github.com/hashicorp/terraform-provider-google/issues/14769
